src/app/work-item/work-item-list/work-item-detail/work-item-detail.component.ts

```diff
@@ -90,19 +91,24 @@ export class WorkItemDetailComponent implements OnInit {
   getAllUsers() {
     this.userService.getAllUsers()
       .then((users) => {
+        //set the creator/reporterUser
```

Comment:
I would move this chunk after line number 99, and use **this.users** instead **users** from the response. That way in future if anyone update **this.reporterUser** anywhere in the component, it'll not update the main item in the service.
Else, this is also fine.